libavfilter/dnn/dnn_backend_openvino.c

```diff
@@ -184,7 +165,7 @@ static DNNReturnType fill_model_input_ov(OVModel *ov_model, RequestItem *request
         request->inferences[i] = inference;
         request->inference_count = i + 1;
         task = inference->task;
-        switch (task->ov_model->model->func_type) {
+        switch (ov_model->model->func_type) {
```

Comment:
you may extract it into a new separate small patch, so I can have an easy life for review.  I mean all the changes 
from task->ov_model to ov_model.

Reply:
Sure. I divide the changes in the OpenVino backend into separate smaller patches.

Currently this patch had the following changes:
1. Extract TaskItems and InferenceItems to `dnn_backend_common.h`
2. Change model in TaskItem to `void *model` and the task->ov_model to ov_model part. (since `task->ov_model` will be a void pointer, so its properties like `task->ov_model->model` cannot be accessed directly without casting it to `((OVModel *) task->ov_model)->model`.
3. Change `char *output_name` to `char **output_names` and corresponding changes in `dnn_backend_openvino.c`

Reply:
But I'm wondering each OVModel in a task points to the same DNNModel instance. So why use `task->ov_model->model` instead of `ov_model` (like ov_model parameter in `fill_model_input_ov`)?

Reply:
I might forget ov_model when I wrote the code.

Reply:
No problem. I'll add those changes as well.